src/NzbDrone.Core/Configuration/ConfigFileProvider.cs

```diff
@@ -216,6 +223,36 @@ public string UrlBase
             }
         }
 
+        public UseProxy ProxyEnabled
+        {
+            get { return GetValueEnum<UseProxy>("ProxyEnabled", UseProxy.Off); }
+        }
+
+        public ProxyType ProxyType
+        {
+            get { return GetValueEnum<ProxyType>("ProxyType", ProxyType.Http); }
+        }
+
+        public string ProxyHostname
+        {
+            get { return GetValue("ProxyHostname", "localhost"); }
```

Comment:
Do we want to store this in the config file or config DB? If we plan to use it in Common I think it needs to live in the file, but the file is less flexible.